lib/improver/cli/standardise.py

```diff
@@ -32,181 +32,41 @@
-                        nargs="+", type=str, default=None,
-                        help="List of names of scalar coordinates to be "
-                             "removed from the non-standard input.")
-    parser.add_argument("--new_name", metavar="NEW_NAME", type=str,
-                        default=None, help="New dataset name.")
-    parser.add_argument("--new_units", metavar="NEW_UNITS", type=str,
-                        default=None, help="Units to convert to.")
-
-    args = parser.parse_args(args=argv)
-
-    if args.target_grid_filepath or args.json_file or args.fix_float64:
-        if not args.output_filepath:
-            msg = ("An argument has been specified that requires an output "
-                   "filepath but none has been provided")
-            raise ValueError(msg)
-
-    if (args.input_landmask_filepath and
-            "nearest-with-mask" not in args.regrid_mode):
-        msg = ("Land-mask file supplied without appropriate regrid_mode. "
-               "Use --regrid_mode=nearest-with-mask.")
-        raise ValueError(msg)
-
-    if args.input_landmask_filepath and not args.target_grid_filepath:
-        msg = ("Cannot specify input_landmask_filepath without "
-               "target_grid_filepath")
-        raise ValueError(msg)
-
-    # Load Cube and json
-    attributes_dict = load_json_or_none(args.json_file)
-    # source file data path is a mandatory argument
-    output_data = load_cube(args.source_data_filepath)
-    target_grid = None
-    source_landsea = None
-    if args.target_grid_filepath:
-        target_grid = load_cube(args.target_grid_filepath)
-        if args.regrid_mode in ["nearest-with-mask"]:
-            if not args.input_landmask_filepath:
-                msg = ("An argument has been specified that requires an input "
-                       "landmask filepath but none has been provided")
-                raise ValueError(msg)
-            source_landsea = load_cube(args.input_landmask_filepath)
-
-    # Process Cube
-    output_data = process(output_data, target_grid, args.regrid_mode,
-                          args.extrapolation_mode, source_landsea,
-                          args.landmask_vicinity, args.regridded_title,
-                          attributes_dict, args.coords_to_remove,
-                          args.new_name, args.new_units, args.fix_float64)
-
-    # Save Cube
-    if args.output_filepath:
-        save_netcdf(output_data, args.output_filepath)
-
-
-def process(output_data, target_grid=None, regrid_mode='bilinear',
-            extrapolation_mode='nanmask', source_landsea=None,
-            landmask_vicinity=25000, regridded_title=None,
-            attributes_dict=None, coords_to_remove=None, new_name=None,
-            new_units=None, fix_float64=False):
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(source_data: cli.inputcube,
+            *,
+            target_grid: cli.inputcube = None,
```

Comment:
(I think `target_grid` and `source_landmask` should be optional positional arguments, so that we can neatly pass the ancils in as inputs in the suite)

Reply:
This I will need to discuss with you, as I had some difficulty getting the `*` in the right place.  I'll address Stephen's comments and then follow up with you.

Reply:
Looks good now, thanks.